TESTS/mbed_hal/critical_section/main.cpp

```diff
@@ -0,0 +1,84 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "critical_section_test.h"
+#include "hal/critical_section_api.h"
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+#include "mbed.h"
+#include "cmsis.h"
+#ifdef TARGET_NRF5 // for all NRF5x targets
+#include "nrf_nvic.h" // for __NRF_NVIC_APP_IRQS_0 / __NRF_NVIC_APP_IRQS_1
+#endif
+
+using utest::v1::Case;
+
+bool test_are_interrupts_enabled(void)
+{
+    // NRF5x targets don't disable interrupts when in critical section, instead they mask application interrupts this is due to BLE stack
+    // (BLE to be operational requires some interrupts to be always enabled)
+#ifdef TARGET_NRF52_DK
```

Comment:
wasn't there somewhere a request for having function to check if interrupts are enabled? Or even an implementation?

Reply:
but it would be good toindeed, there is function `hal_in_critical_section()`, but it would be nice to verify API functions by something form outside critical section API. `us_ticker` could be used to check interrupts but unfortunately it utilises `core_util_critical_section_` (which also call HAL critical API ) and it brakes the test. Generly we should always use platform `core_util_critical_section` not HAL, in particular we cannot use both in the same time

Reply:
I was concerned about the portability (how this function can grow in time).

Looks fine to test this here.